extended/src/main/java/io/kubernetes/client/extended/leaderelection/resourcelock/LeaseLock.java

```diff
@@ -135,15 +135,23 @@ private LeaderElectionRecord getRecordFromLease(V1LeaseSpec lease) {
   }
 
   private V1LeaseSpec getLeaseFromRecord(LeaderElectionRecord record) {
-    return new V1LeaseSpec()
-        .acquireTime(
-            OffsetDateTime.ofInstant(
-                Instant.ofEpochMilli(record.getAcquireTime().getTime()), ZoneOffset.UTC))
-        .renewTime(
-            OffsetDateTime.ofInstant(
-                Instant.ofEpochMilli(record.getRenewTime().getTime()), ZoneOffset.UTC))
-        .holderIdentity(record.getHolderIdentity())
-        .leaseDurationSeconds(record.getLeaseDurationSeconds())
-        .leaseTransitions(record.getLeaderTransitions());
+    V1LeaseSpec spec =
+        new V1LeaseSpec()
+            .holderIdentity(record.getHolderIdentity())
+            .leaseDurationSeconds(record.getLeaseDurationSeconds())
+            .leaseTransitions(record.getLeaderTransitions());
+    if (record.getAcquireTime() != null) {
+      spec =
+          spec.acquireTime(
+              OffsetDateTime.ofInstant(
+                  Instant.ofEpochMilli(record.getAcquireTime().getTime()), ZoneOffset.UTC));
```

Comment:
This drops the milliseconds, right? Is that the intention? It also assumes that the timestamps are all UTC (but I think that is guaranteed by the k8s spec if they come from the server).

Reply:
yes, the leader-elector manages resource-locks w/ millis-sec timestamp, so it doesn't make a difference whether `LeaseLock` provides extra precision.